src/test/kotlin/com/zepben/evolve/database/sqlite/DatabaseSqliteTest.kt

```diff
@@ -259,6 +265,50 @@ class DatabaseSqliteTest {
         validateSchema(SchemaNetworks.createNameTestServices())
     }
 
+    @Test
+    internal fun `post process fails with unresolved references`() {
+        val builder = TestNetworkBuilder()
+        val ns = builder.network
+        builder
+            .fromPowerTransformer() // b0
+            .toPowerElectronicsConnection {  // pec1
+                TapChangerControl().also { tcc ->
+                    tcc.addRegulatingCondEq(this)
+                    tcc.terminal = Terminal("test").also { ns.add(it) }
+                    regulatingControl = tcc
+                    ns.add(tcc)
+                }
+            }
+            .build()
+
+        val expectedMetadata = MetadataCollection()
+        val expectedDiagramService = DiagramService()
+        val expectedCustomerService = CustomerService()
+
+        assertThat(
+            DatabaseWriter(SCHEMA_TEST_FILE).save(
+                expectedMetadata,
+                mutableListOf(
+                    ns,
+                    expectedDiagramService,
+                    expectedCustomerService
+                )
+            ),
+            equalTo(true)
+        )
+
+        val metadataCollection = MetadataCollection()
+        val networkService = NetworkService()
+        val diagramService = DiagramService()
+        val customerService = CustomerService()
+
+        // This will throw if unresolved references still exist after a full load
+        assertThat(
```

Comment:
**todo:** This is not a good test because it doesn't test that the throw happens. Someone could just remove the code that throws on unresolved references and you'd be none the wiser. Surely if you want to test that bit of code you remove the object in the database before loading so the resolver fails?

Reply:
yep didn't think of that - fixed.